### DISCUSSION TITLE
Are you taking suggestions for small demos?

### DISCUSSION THREAD

**Discussion by goodtothinkwith** [daveshap/ACE_Framework#17] [General]
As a couple of quick, text-based knowledge tasks that would be incredibly useful just to kick things off:

1. Research and write a scientific paper on a particular topic. (Even if it’s just the lit review, that would be crazy useful)
2. Given a particular function (like an LLM API call) and an evaluation function to score the output, design and run an experiment. 

The second one is structured like an optimization routine. If that works, it would provide a general solution for finding the best way of doing something with an LLM… and that could then be used to improve the ACE model, among other things. 

I hope this helps and catalyzes some productive thought :)

**Comment by daveshap** [comment]
The core team is building an MVP/POC that will be hackable, Beyond that, we are focusing on the Triple-E (educate, empower, and enable) - we are going to enable, empower, and educate other teams to build what they want to.

**Reply by bitplane** [reply]
I'll dump my idea here: create an agent that can solve an arbitrary interactive fiction game.

There's constraints on outputs (to drive the game) but not on inputs (outputs from the game), exploration of a model of the virtual world derived from text, and to make decisions with temporal aspects, and to keep notes on and learn from previous playthroughs. Also depending on the game the best response might be that the agent fails (e.g. the game requires unethical actions in order to complete it, which the agent refuses to do and so can't succeed).

There's a ton of free IF games and even more abandonware that can be used with emulators and screen scrapers/OCR so it's quite a practical challenge. OCR would be a useful component for creating internal models of things. Navigating virtual environments is useful in general, so adding these components would apply broadly across all problems. It also might fit with the ACE model as the "text to telemetry" issue would need to be solved too
